semantics/cpp/language/common/dynamic.k

```diff
@@ -1076,6 +1077,8 @@ module CPP-DYNAMIC-OTHER
 
      rule idOf(_ :: S::ClassSpecifier) => idOf(S)
 
+     rule baseOfLoc(loc(Base::SymBase, _) #Or loc(Base::SymBase, _, _)) => Base
```

Comment:
I think there's already a `base(SymLoc)` function for this in `common/symloc.k`.